src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void validIndexCheck(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index: " + index);
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            head = node;
+            tail = head;
+        } else {
+            node = new Node<>(tail,value,null);
+            if (tail.prev == null) {
+                tail.prev = head;
+                head.next = node;
+            } else {
+                tail.next = node;
+                node.prev = tail;
+            }
+            tail = node;
```

Comment:
There is a logical error in the else block. You should not create a new node again (as you did at line 31). Instead, create mutual links from both sides when you insert a new Node. Also, tail.prev will never be null if size is not 0, so the check and assignment at lines 37-39 are unnecessary.